Problem1_PrimeTime.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Net.Sockets;
+using System.Net;
+using System.IO.Pipelines;
+using System.Text.Json;
+using System.Buffers;
+using System.Text.Unicode;
+using System.Text;
+
+namespace ProtoHackers
+{
+    class Problem1_PrimeTime
+    {
+        public static async Task Init(int port)
+        {
+            var listenSocket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+            listenSocket.Bind(new IPEndPoint(IPAddress.IPv6Any, port));
+
+            listenSocket.Listen();
+
+            while (true)
+            {
+                var socket = await listenSocket.AcceptAsync();
+                _ = Handle(socket);
+            }
+        }
+
+        // largely copied from https://learn.microsoft.com/en-us/dotnet/standard/io/pipelines#pipe-basic-usage
+        static async Task Handle(Socket socket)
+        {
+            await using var stream = new NetworkStream(socket, true);
+            var reader = PipeReader.Create(stream);
+
+            try
+            {
+                while (true)
+                {
+                    ReadResult result = await reader.ReadAsync();
+                    ReadOnlySequence<byte> buffer = result.Buffer;
+
+                    while (TryReadLine(ref buffer, out ReadOnlySequence<byte> line))
+                    {
+                        // Process the line.
+                        var handleResult = await HandleMessage(line, stream);
+                        if (!handleResult)
+                        {
+                            throw new Exception();
```

Comment:
Why throw here instead of responding? Are you supposed to end the connection on any error?

Reply:
Yeah, I don't like how the control flow is right now. From the problem:
>A request is malformed if it is not a well-formed JSON object, if any required field is missing, if the method name is not "isPrime", or if the number value is not a number.

and

>Whenever you receive a malformed request, send back a single malformed response, and disconnect the client.

The rationale of throwing was to unify the handling of the invalid JSON case (an exception thrown by `JsonDocument.Parse`) and the other field validation.

I updated to use `JsonDocument.TryParseValue` in `HandleMessage` to avoid using exceptions there to communicate "malformed". That seems a little nicer, but there is still the exception in the higher level handler just as a way to exit the processing loop. I am not sure the best way to "disconnect" at least with the way the code is currently structured.

Reply:
Updated to use a boolean for control flow. It still feels clunky, but no more exceptions